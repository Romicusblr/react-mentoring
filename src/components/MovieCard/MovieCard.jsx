/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import MovieYear from '../MovieYear';
import List from '../List';
import style from './MovieCard.module.css';

function MovieCard({
  movie, className, ...other
}) {
  if (!movie) return null;

  const {
    id, title, genres, release_date, poster_path,
  } = movie;

  return (
    <Link to={`/film/${id}`}>
      <figure
        className={classNames(style.movieCard, className)}
        {...other}
      >
        <img src={poster_path} alt={title} />
        <figcaption>
          <h3 className={style.title}>
            {title}
            <MovieYear year={new Date(release_date).getFullYear()} />
          </h3>
          <List items={genres} limit={2} delimeter=" &amp; " />
        </figcaption>
      </figure>
    </Link>
  );
}

MovieCard.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.number,
    poster_path: PropTypes.string,
    title: PropTypes.string,
    genres: PropTypes.arrayOf(PropTypes.string),
    release_date: PropTypes.string,
  }),
  className: PropTypes.string,
};

MovieCard.defaultProps = {
  movie: null,
  className: '',
};

export default MovieCard;
