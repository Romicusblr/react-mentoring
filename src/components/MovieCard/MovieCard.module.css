.movieCard {
  width: 20em;
  margin: 0;
  margin-bottom: 2rem;
  &>img {
    width: 100%;
  }
}

.title {
  margin-bottom: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  font-weight: normal;
  color: var(--color-text-secondary)
}
