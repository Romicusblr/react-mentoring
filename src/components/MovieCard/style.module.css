.movieCard {
    width: 20em;

    &>figcaption {
        &>div {
            display: flex;
            align-items: center;
            justify-content: space-between;
        }
    }
}
