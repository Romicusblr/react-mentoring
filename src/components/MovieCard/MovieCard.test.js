import MovieCard from '.';

fdescribe('MovieCard', () => {
  it('should be selectable by class "movieCard"', () => {
    expect(shallow(<MovieCard />).is('.movieCard')).toBe(true);
  });

  it('should match to snapshot', () => {
    const snap = shallow(<MovieCard />);
    expect(snap).toMatchSnapshot();
  });
});
