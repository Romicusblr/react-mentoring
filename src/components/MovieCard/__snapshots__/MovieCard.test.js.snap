// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match to snapshot 1`] = `
<figure
  className="movieCard"
>
  <img />
  <figcaption>
    <h3
      className="title"
    >
      <MovieYear
        className=""
        year={NaN}
      />
    </h3>
    <List
      className=""
      delimeter=" & "
      items={Array []}
      limit={2}
    />
  </figcaption>
</figure>
`;
